src/Misc/tools.ts

```diff
@@ -938,11 +938,10 @@ export class Tools {
      * @param url the input url
      * @return the absolute url
      */
-    public static GetAbsoluteUrl(url: string): string {
-        const a = document.createElement("a");
-        a.href = url;
-        return a.href;
-    }
+    public static GetAbsoluteUrl: (url: string) => string =
+        (typeof document === "object") ? (url) => { const a = document.createElement("a"); a.href = url; return a.href; } :
+        (typeof URL === "function") ? (url) => new URL(url, location.origin).href :
+        (url) => `${location.origin}/${url}`;

```

Comment:
This will still fail if `location` is not available (like in a node.js scenario), but people can simply override this function if it needs to work.

Reply:
let s return url by default instead of a crash ?

Reply:
I thought about that, but is that good? It may return a relative url when the function is called (GetAbsoluteUrl). Another thought I had is to add a static variable (e.g., BaseUrl that can be used to as a fallback or maybe override).

Reply:
Makes sense the static one could be prefixed if it does not already start with http... I kind of like it :-)

Reply:
You mean as a fallback or override? Also, what should the default of the BaseUrl be? :-)

Reply:
As you say, it is possible to override this function.  Maybe a warning would be good - if no location is available, warn that it is not there and that this function can be adjusted.
Otherwise, return `/` as the base URL, IMO it makes the most sense.

Reply:
I forgot that just using string concat is wrong. If the incoming URL is already absolute, this will result in a bad URL. Detecting whether a URL is absolute or relative is not trivial. I think we should perhaps just throw an error for the fallback case with a message saying you can override the function?

Reply:
I think this is fine. Just log the error